unittest/quaternion.cpp

```diff
@@ -42,5 +42,24 @@ BOOST_AUTO_TEST_CASE(test_uniformRandom)
   }
 }
 
+BOOST_AUTO_TEST_CASE(test_isNormalized)
+{
+  srand(0);
+
+  using namespace pinocchio;
+  typedef Eigen::Quaternion<double> Quaternion;
+  typedef Quaternion::Coefficients Vector4;
+  
+  for(int i = 0; i < 1e6; ++i)
+  {
+    Quaternion q;
+    q.coeffs() = Vector4::Random() + Vector4::Constant(2);
+    BOOST_CHECK(!quaternion::isNormalized(q));
+    
+    q.normalize();
+    BOOST_CHECK(quaternion::isNormalized(q));
```

Comment:
Looks great - thank you for the quick fix. We could also add an explicit test against Vector4d::Zero()

Reply:
Done